pysmt/simplifier.py

```diff
@@ -664,3 +666,73 @@ def walk_div(self, formula, args, **kwargs):
         return self.manager.Div(sl, sr)
 
 # EOC Simplifier
+
+
+class BddSimplifier(Simplifier):
+    def __init__(self, env=None, static_ordering=None, bool_abstraction=False):
+        Simplifier.__init__(self, env=env)
+        self._validation_sname = None
+
+        Solver = self.env.factory.Solver
+        if static_ordering is not None:
+            self.s = Solver(name="bdd", static_ordering=static_ordering)
+        else:
+            self.s = Solver(name="bdd", dynamic_reordering=True)
+        self.convert = self.s.converter.convert
+        self.back = self.s.converter.back
+        # Set methods for boolean_abstraction
+        self.bool_abstraction = bool_abstraction
+        self.set_function(self.walk_abstract, *op.RELATIONS)
+        self.set_function(self.walk_abstract, op.FUNCTION)
+        self.ba_map = {}
+        self.get_type = self.env.stc.get_type
+        self.FreshSymbol = self.env.formula_manager.FreshSymbol
+
+    @property
+    def validate_simplifications(self):
+        return self._validate_simplifications
+
+    @validate_simplifications.setter
+    def validate_simplifications(self, value):
+        possible_solvers = [sname for sname in self.env.factory.all_solvers()\
+                            if sname!="bdd"]
+        if len(possible_solvers) == 0:
+            raise ValueError("To validate at least another solver must be available!")
+        self._validation_sname = possible_solvers[0]
+        self._validate_simplifications = value
+
+    def simplify(self, formula):
+        from pysmt.oracles import get_logic
+        from pysmt.logics import BOOL, QF_BOOL
+        if self.bool_abstraction:
+            logic = get_logic(formula)
+            if logic > QF_BOOL and logic != BOOL:
+                res = self.abstract_and_simplify(formula)
+            else:
+                res = self.back(self.convert(formula))
+        else:
+            res = self.back(self.convert(formula))
+        self._validate(formula, res)
+        return res
+
+    def _validate(self, old, new):
+        if self.validate_simplifications:
+            Iff = self.env.formula_manager.Iff
+            is_valid = self.env.factory.is_valid
+            sname = self._validation_sname
+            assert is_valid(Iff(old, new), solver_name=sname ), \
+              "Was: %s \n Obtained: %s\n" % (str(old), str(new))
+
+    def abstract_and_simplify(self, formula):
+        abs_formula = self.walk(formula)
```

Comment:
Even if the Simplifier recognizes that two atoms are equivalent the BDD could be reduced...